core/src/test/java/org/infinispan/util/CoreTestBlockHoundIntegration.java

```diff
@@ -0,0 +1,86 @@
+package org.infinispan.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+import org.infinispan.commons.internal.CommonsBlockHoundIntegration;
+import org.infinispan.commons.test.PolarionJUnitXMLWriter;
+import org.infinispan.distribution.BlockingInterceptor;
+import org.infinispan.eviction.impl.EvictionWithConcurrentOperationsTest;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.test.concurrent.StateSequencer;
+import org.infinispan.test.fwk.CheckPoint;
+import org.infinispan.test.fwk.TestResourceTracker;
+import org.infinispan.util.concurrent.ReclosableLatch;
+import org.infinispan.util.logging.LogFactory;
+import org.kohsuke.MetaInfServices;
+
+import io.reactivex.exceptions.UndeliverableException;
+import reactor.blockhound.BlockHound;
+import reactor.blockhound.integration.BlockHoundIntegration;
+
+@SuppressWarnings("unused")
+@MetaInfServices
+public class CoreTestBlockHoundIntegration implements BlockHoundIntegration {
+   @Override
+   public void applyTo(BlockHound.Builder builder) {
+      allowTestsToBlock(builder);
+
+      builder.allowBlockingCallsInside(CoreTestBlockHoundIntegration.class.getName(), "writeJUnitReport");
+
+      builder.blockingMethodCallback(bm ->
+            writeJUnitReport(new AssertionError(String.format("Blocking call! %s on thread %s", bm, Thread.currentThread())), "Blocking"));
```

Comment:
Please also do a `TestSuiteProgress.fakeTestFailure(testName, exception)`, I have a script that looks for those `Test failed` messages in the output when I run the tests locally :)